docs/source/programming.md

```diff
@@ -0,0 +1,123 @@
+:::
+
+#### 2.3. Interaction Hamiltonian
+
+
+The interaction Hamiltonian depends on the distance between the atoms $i$ and $j$, $R_{ij}$, and the energy levels in which the information is encoded in these atoms, that define the interaction between the atoms $\hat{U}_{ij}$
+
+$$
+H^\text{int}_{ij} = \hat{U}_{ij}(R_{ij})
+$$
+
+The interaction operator $\hat{U}_{ij}$ is composed of an entangling operator and an interaction strength:
+- If the Rydberg state $\left|r\right>$ is involved in the computation, then $\hat{U}_{ij}(R_{ij}) = \frac{C_6}{R_{ij}^6} \hat{n}_i \hat{n}_j$, where the interaction strength is $\frac{C_6}{R_{ij}^6}$, with $C_6$ a coefficient that depends on the principal quantum number of the Rydberg state. The entangling operator between atom $i$ and $j$ is $\hat{n}_i\hat{n}_j = |r\rangle\langle r|_i |r\rangle\langle r|_j$. Together with the driving Hamiltonian, this interaction encodes the _Ising Hamiltonian_ and is the **most common choice in neutral-atom devices.**
+- If the information is stored in the $\left|0\right>$ and $\left|1\right>$ states of the `XY` basis, $\hat{U}_{ij}(R_{ij}) =\frac{C_3}{R_{ij}^3} (|1\rangle\langle 0|_i |0\rangle\langle 1|_j + |0\rangle\langle 1|_i |1\rangle\langle 0|_j)$, where the interaction strength is $\frac{C_3}{R_{ij}^3}$, with $C_3$ a coefficient that depends on the energy levels used to encode $\left|0\right>$ and $\left|1\right>$. The entangling operator between atom $i$ and $j$ is $\hat{\sigma}_i^{+}\hat{\sigma}_j^{-} + \hat{\sigma}_i^{-}\hat{\sigma}_j^{+} = |1\rangle\langle 0|_i |0\rangle\langle 1|_j + |0\rangle\langle 1|_i |1\rangle\langle 0|_j$. This interaction hamiltonian is associated with the _XY Hamiltonian_ and is a less common mode of operation, usually accessible only in select neutral-atom devices.
+
+:::{important}
+With Pulser, you program the interaction Hamiltonian by setting the distance between atoms, $R_{ij}$, and the choice of eigenstates used in the computation and the Rydberg level(s) targeted by the device. Most commonly, the ground and Rydberg states ($\left|g\right>$ and $\left|r\right>$) are used, such that
+$$
+H^\text{int}_{ij} = \frac{C_6}{R_{ij}^6} \hat{n}_i\hat{n}_j
+$$
+When providing the distance between atoms, Pulser ensures that you respect the constraints of your chosen device.
+:::
+
+#### 2.4. Evolution of the system's state
+
+For a system of atoms in the initial state $\left|\Psi_0\right>$, the final state of the system after a time $\Delta t$ is:
+$$ \left|\Psi_f\right> = \exp\left(-\frac{i}{\hbar}\int_0^{\Delta t} H(t) dt\right)$$
+
+## Writing a Pulser program
+
+As outlined above, Pulser lets you program an Hamiltonian so that you can manipulate the quantum state of a system of atoms, that encode the states. The series of necessary instructions is encapsulated in the so-called Pulser `Sequence`. Here is a step-by-step guide to create your own Pulser `Sequence`.
```

Comment:
an Hamiltonian -> the Hamiltonian presented above.

so that you -> so that the user 

I would maybe delete "that encode the states" as redundant and explained above